connectors/src/connectors/notion/temporal/workflows.ts

```diff
@@ -89,26 +100,45 @@ export async function notionSyncWorkflow(
     do {
       const { pageIds, nextCursor } = await notionGetPagesToSyncActivity(
         notionAccessToken,
-        lastSyncedPeriodTs,
+        // if we're doing a garbage collection run, we want to fetch all pages
+        !isGargageCollectionRun ? lastSyncedPeriodTs : null,
         cursor,
         {
           pageIndex,
           dataSourceName: dataSourceConfig.dataSourceName,
           workspaceId: dataSourceConfig.workspaceId,
+          runType: isGargageCollectionRun
+            ? "garbageCollection"
+            : isInitialSync
+            ? "initialSync"
+            : "incrementalSync",
         }
       );
       cursor = nextCursor;
       pageIndex += 1;
 
-      const pagesToSync = pageIds.filter(
-        (pageId) => !pagesSyncedWithinPeriod.has(pageId)
```

Comment:
got rid of this "optimization": we sync every ~10s, but notion TS is only precise to the minute, which means when a page changes we might end up upserting it up to 6 times. I think it is better than potentially missing changes (eg, the page changed multiple times during a minute, and stops updating before the end of the minute).

If we end up upserting too much, we can consider increasing the interval between our sync runs